.github/workflows/rust-container.yaml

```
@@ -0,0 +1,47 @@
+# GitHub Actions workflow for building and pushing Rust utilities container
+# Rebuilds when Rust code changes (utils/**, Cargo.toml, Cargo.lock)
+
+name: Build Rust Utilities Container
+
+on:
+  push:
+    branches: ['master', 'dev']
+    paths:
+      - 'utils/**'
+      - 'Cargo.toml'
+      - 'Cargo.lock'
+      - 'containers/rust-utils/**'
+      - '.github/workflows/rust-container.yaml'
+  workflow_dispatch:
+    inputs:
+      force_rebuild:
+        description: 'Force rebuild Rust utilities container'
+        required: false
+        type: boolean
+        default: false
+
+jobs:
+  build-rust-utils:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Docker login
+        env:
+          DOCKER_USER: ${{ secrets.DOCKER_USER }}
+          DOCKER_PASSWORD: ${{ secrets.DOCKER_PASSWORD }}
+        run: docker login -u $DOCKER_USER -p $DOCKER_PASSWORD
+
+      - name: Build Rust utilities container
```

The workflow will continue pushing even if the build fails. Consider adding:
```suggestion
      - name: Build Rust utilities container
        if: success()  # Only push if build succeeded
```

---

With GHA, the whole pipeline will fail if one step fails, so we shouldn't end up pushing if the build fails.